src/archival/utils.rs

```diff
@@ -59,6 +79,51 @@ pub async fn is_row_exists(pool: &PgPool, row_id: i32) -> bool {
     }
 }
 
+pub async fn make_archival_network_request(
+    url: &str,
+) -> Result<String, Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
`dyn Error`s are a pretty bad practice, as it forces complicated return types. If you don't care about getting the type of the returned error, you can use the `color_eyre` crate's `Result` that will abstract the error for you.

If you do care about keeping the error type, you can create an error struct yourself that incapsulate both serde and reqwest error. Since it's a lot of boilerplate, I recommend the crate `this_error` to create this struct. Here's an example: https://github.com/RustyNova016/listenbrainz-cli-tools/blob/master/src/models/error.rs